bfapi/service/jobs.py

```diff
@@ -221,13 +225,12 @@ def __genCmd(
                     '--tolerance 0.075',
                     'shoreline.geojson',
                 ])
-    elif svc_format == "pzsvc-ndwi-py":
+    elif algo_interface == "pzsvc-ndwi-py":
         return ' '.join(["--b1", geotiff_filenames[0], "--b2", geotiff_filenames[1], "--fout ./shoreline.json"])
-    elif svc_format == "test-algo-interface":
-        return "algo-test-command"
     else:
-        log.error('cmd string requested for unknown service type')
-        raise Exception()
+        error_string = 'CLI cmd string requested for unknown algorithm interface "' + algo_interface + '".'
+        log.error(error_string)
+        raise PreprocessingError(message=error_string)
```

Comment:
Avoid using data types in variable names, ala `error_string`.  `error_message` is more conventional and conveys more meaning.

I also should have mentioned this earlier, but `PreprocessingError` gets serialized and sent with the HTTP response so it needs to be cleaned up.  This message can probably be shortened to `unknown algorithm interface 'foo'`.